ouroboros-consensus-test/bench-storage/Bench/Ouroboros/Consensus/Storage/LedgerDB/HD/DiffSeq.hs

```diff
@@ -961,41 +958,86 @@ sanityCheck bEnv = conjoin [
-    , isomorphismProperties2
     ]
   where
     BenchEnv {cmdsAntidiff, cmdsIntermediateSum, ds0Antidiff, ds0IntermediateSum, model} = bEnv
 
     interpretationsMatch = counterexample "interpret matches" $
-      interpret ds0Antidiff cmdsAntidiff === to (interpret ds0IntermediateSum cmdsIntermediateSum)
+        interpret ds0Antidiff cmdsAntidiff
+      ===
+        fromSeqUtxoDiff (interpret ds0IntermediateSum cmdsIntermediateSum)
 
     modelMatchesInterpretation1 = counterexample "model matches interpretations 1" $
       diffs model === interpret ds0Antidiff cmdsAntidiff
 
     modelMatchesInterpretation2 = counterexample "model matches interpretations 2" $
-       diffs model === to (interpret ds0IntermediateSum cmdsIntermediateSum)
+        diffs model
+       ===
+        fromSeqUtxoDiff (interpret ds0IntermediateSum cmdsIntermediateSum)
 
     lengthsMatch1 = counterexample "Lengths of command/diff sequences match 1" $
       length cmdsAntidiff === length cmdsIntermediateSum
 
     lengthsMatch2 = counterexample "Lengths of command/diff sequences match 2" $
       DS.length ds0Antidiff === HD.lengthSeqUtxoDiff ds0IntermediateSum
 
-    isomorphismCheck1 = counterexample "Isomorphism checks 1" $
-      cmdsAntidiff === to cmdsIntermediateSum .&&.
-      ds0Antidiff === to ds0IntermediateSum
-
-    isomorphismCheck2 = counterexample "Isomorphism checks 2" $
-      to cmdsAntidiff === cmdsIntermediateSum .&&.
-      to ds0Antidiff === ds0IntermediateSum
-
-    isomorphismProperties1 = counterexample "Isomorphism properties 1" $
-      isomorphismLaw (Proxy @[Cmd HD.SeqUtxoDiff k v]) cmdsAntidiff .&&.
-      isomorphismLaw (Proxy @(HD.SeqUtxoDiff k v)) ds0Antidiff
+{-------------------------------------------------------------------------------
+  Conversion
+-------------------------------------------------------------------------------}
 
-    isomorphismProperties2 = counterexample "Isomorphism properties 2" $
-      isomorphismLaw (Proxy @[Cmd DS.DiffSeq k v]) cmdsIntermediateSum .&&.
-      isomorphismLaw (Proxy @(DS.DiffSeq k v)) ds0IntermediateSum
+fromDiffSeqCmd :: Cmd DS.DiffSeq k v1 -> Cmd HD.SeqUtxoDiff k v1
+fromDiffSeqCmd cmd = case cmd of
+  Push sl d     -> Push sl (fromDiff d)
+  Flush n       -> Flush n
+  Switch n ps   -> Switch n (fmap (second fromDiff) ps)
+  Forward vs ks -> Forward (fromValues vs) (fromKeys ks)
+
+fromDiffSeq :: Ord k => DS.DiffSeq k v -> HD.SeqUtxoDiff k v
+fromDiffSeq (DS.UnsafeDiffSeq ft) = HD.SeqUtxoDiff . FT.fromList . map fromElement . toList $ ft
+
+fromElement :: DS.Element k v -> HD.SudElement k v
+fromElement (DS.Element slot d) = HD.SudElement slot (fromDiff d)
+
+fromDiff :: DS.Diff k v -> HD.UtxoDiff k v
+fromDiff (MapDiff.Diff m) = HD.UtxoDiff (fmap fromNEDiffHistory m)
+
+fromNEDiffHistory :: DS.NEDiffHistory v -> HD.UtxoEntryDiff v
+fromNEDiffHistory (MapDiff.NEDiffHistory neseq) = case neseq of
+  Empty :|> x :||> y -> fromDiffEntry x <> fromDiffEntry y
```

Comment:
Minor: maybe aligning this will make this definition more readable.

Reply:
Done